lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -1137,11 +1137,14 @@ void OCLToSPIRVBase::visitCallRelational(CallInst *CI,
   OCLSPIRVBuiltinMap::find(DemangledName.str(), &OC);
   // i1 or <i1 x N>, depending on whether it returns a vector type.
   Type *BoolTy = CI->getType()->getWithNewType(Type::getInt1Ty(*Ctx));
-  mutateCallInst(CI, OC).changeReturnType(BoolTy, [=](IRBuilder<> &Builder,
-                                                      CallInst *NewCI) {
-    return Builder.CreateSelect(NewCI, Constant::getAllOnesValue(CI->getType()),
-                                Constant::getNullValue(CI->getType()));
-  });
+  mutateCallInst(CI, OC).changeReturnType(
+      BoolTy, [=](IRBuilder<> &Builder, CallInst *NewCI) {
+        Value *TruOp = CI->getType()->isVectorTy()
+                           ? Constant::getAllOnesValue(CI->getType())
+                           : getInt32(M, 1);
+        return Builder.CreateSelect(NewCI, TruOp,
+                                    Constant::getNullValue(CI->getType()));
```

Comment:
guess, there is a typo
```suggestion
      BoolTy, [=](IRBuilder<> &Builder, CallInst *NewCI) {
        Value *TrueOp = CI->getType()->isVectorTy()
                           ? Constant::getAllOnesValue(CI->getType())
                           : getInt32(M, 1);
        return Builder.CreateSelect(NewCI, TrueOp,
                                    Constant::getNullValue(CI->getType()));
```